File: src/2425. Bitwise XOR of All Pairings.py
class Solution:
    def xorAllNums(self, nums1: List[int], nums2: List[int]) -> int:
        x = y = 0
        for num in nums1:
            x ^= num
        for num in nums2:
            y ^= num
        return (len(nums1) % 2 * y) ^ (len(nums2) % 2 * x)

File: src/2425. Bitwise XOR of All Pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int x = 0, y = 0;
        for (int num : nums1)
            x ^= num;
        for (int num : nums2)
            y ^= num;
        return (nums1.size() % 2 * y) ^ (nums2.size() % 2 * x);
    }
};